### DISCUSSION TITLE
Add client's name to logging output

### DISCUSSION THREAD

**Discussion by Andy081549** [jamulussoftware/jamulus#3447] [Ideas]
Hello all,

I unfortunately have only basic programming skills and this is beyond my skills...

I'd really like to see which user is connecting with which ip address. It seems that the logging string is generated in  serverlogging.cpp (line 52). Is it possible to also have the client name in this string?

Kind regards
Andy

**Comment by ann0see** [comment]
For privacy reasons this can only happen on the server. Also there you should already see it from the logs (IP + channel ID) or the JSON-RPC API already.

**Comment by Andy081549** [comment]
The server is where I want to use it but unfortunately it only logs time and ip address.

I just want to exclude some (fortunately there are only very few) people from connecting to my server with iptables. Hope this is not a problem with privacy (I don't think it is, because there "special behaving" people use alias names).

**Reply by ann0see** [reply]
That's a controversial topic unfortunately. The way to go is using JSON-RPC: https://github.com/jamulussoftware/jamulus/blob/main/docs/JSON-RPC.md

The jamulusserver/getClients call is what you want. It will give you the IP (which can surely change!). 

I'd look into how to use JSON-RPC in bash, perl or any language you feel comfortable with.

Just so you know, for privacy reasons, I'd personally even get rid of logging the IPs entirely - but this got rejected years ago.

**Reply by ann0see** [reply]
If you need help feel free to ask. 
I'm quite sure people have already written software for your purposes. 
Jamban for example: https://github.com/dingodoppelt/jamban

**Comment by pcar75** [comment]
A little research in this forum or the previous ( https://sourceforge.net/p/llcon/discussion/ ;  Home Browse Jamulus - Internet Jam Session Software Discussion ) might help.  For example , you may be interested by "Log file modification request #1969".  Some people have developed processes or variants of the official releases to cater for the infrequent cases - but quite annoying and sometime persistent- ill-behaved users.
